app/Http/Controllers/Legacy/Web/ActionsController.php

```diff
@@ -45,7 +46,8 @@ public function store(Request $request)
         $this->validate($request, [
             'name' => 'required|string',
             'campaign_id' => 'required|integer|exists:campaigns,id',
-            'post_type' => 'required|string',
+            'post_type' => 'required|string|in:photo,voter-reg,text,share-social,phone-call',
+            'callpower_campaign_id' => 'integer',
```

Comment:
Let's make sure this is [`required_if`](https://laravel.com/docs/5.5/validation#rule-required-if) the `post_type` is `phone-call`.

Reply:
We also probably want to make sure we don't attribute a single CallPower campaign to more than one action, so let's add an [`exists`](https://laravel.com/docs/5.5/validation#rule-exists) rule for that.

Reply:
great call @DFurnes ! 

With `integer` also in the validation, it was throwing an error when I tried to create other post types, saying that `callpower_campaign_id` had to be an integer (and it was empty). Seems like this is a [known issue](https://github.com/laravel/framework/issues/17032) with some hacky fixes. To get around this, I just removed `integer` validation since we're using the `exists` validation and I think that should work!

Reply:
> We also probably want to make sure we don't attribute a single CallPower campaign to more than one action, so let's add an exists rule for that.

@DFurnes oh shoot - testing this on QA I think I misunderstood what you were saying here! I need to update this validation in a new PR but not sure how to make sure a CallPower campaign only belongs to one action  with the `exists` rule?

Reply:
> With `integer` also in the validation, it was throwing an error when I tried to create other post types, saying that `callpower_campaign_id` had to be an integer (and it was empty). Seems like this is a [known issue](https://github.com/laravel/framework/issues/17032) with some hacky fixes. To get around this, I just removed `integer` validation since we're using the `exists` validation and I think that should work! 

This seems to be working for me. Can you try again?

```php
'post_type' => 'required|string|in:photo,voter-reg,text,share-social,phone-call',
'callpower_campaign_id' => 'required_if:post_type,phone-call|integer'
```

> I need to update this validation in a new PR but not sure how to make sure a CallPower campaign only belongs to one action with the `exists` rule?

Oops, sorry. Meant to link to its opposite – the [`unique`](https://laravel.com/docs/5.5/validation#rule-unique) rule!

Reply:
ah got it! PR [here](https://github.com/DoSomething/rogue/pull/850)!